pkg/app/api/service/pipedservice/service.proto

```diff
@@ -147,6 +148,12 @@ service PipedService {
 
     // ListEvents returns a list of Events inside the given range.
     rpc ListEvents(ListEventsRequest) returns (ListEventsResponse) {}
+
+    // GetMostRecentSuccessfulAnalysisMetadata returns the most successful analysis metadata.
+    rpc GetMostRecentSuccessfulAnalysisMetadata(GetMostRecentSuccessfulAnalysisMetadataRequest) returns (GetMostRecentSuccessfulAnalysisMetadataResponse) {}
```

Comment:
How about just `GetAnalysisMetadata` because we are saving only the latest one in the store?

Reply:
The same reason as https://github.com/pipe-cd/pipe/pull/2483#discussion_r710678222

Reply:
How about at least `GetLatestAnalysisMetadata`? With only `GetAnalysisMetadata()`, it's quite difficult to know what's going on.

Reply:
One thing to confirm, from the caller side, we only expect the latest successful analysis stage metadata, right? In that case `Successful`keyword has meaning to me.

Reply:
Yes. So I honestly want to clarify that. But I know way too long.

Always tough to determine the names

Reply:
Right. Naming is really hard. 🤯 
`GetLatestAnalysis...` looks fine to me.
Because by design, we assume that only successful data will be stored.
(It will be long and hard to put all the meaning words in one thing.)

Reply:
Not sure I have to mention or not but we already have long name functions already so I guess it's not a big deal. Just `not too long` is fine.

Reply:
Cool. It seems to have both readability and the right meaning.

Reply:
>Not sure I have to mention or not but we already have long name functions already so I guess it's not a big deal.

We're on the same page. This is one of the reasons I settled on using such a long name 🙄 
But I got satisfied with the name.

Reply:
> Not sure I have to mention or not but we already have long name functions already so I guess it's not a big deal. Just not too long is fine.

Nice point, I got it. 👍  
We are already having some long names but somethings I want to say here
- firstly, they are in different contexts,  for the datastore case, all data are stored and we have `Get...Successul..Deployment` to explicitly show what filters should be used, what data point will be returned
- secondly, maybe the long names we created before are not good too 😄 . So no reason to add another not-good thing because we are already having some not-good things. Instead of that, we should improve the old names in the future.

In short, IMO I think
- it is not a big deal because long-name is more understandable (agree)
- it is not a big deal because we are already having something like that (not totally agree)